stock_resupply_order/models/stock_resupply_order.py

```diff
@@ -0,0 +1,200 @@
+        tracking=True,
+        store=True,
+        default="draft",
+    )
+
+    picking_ids = fields.One2many(
+        "stock.picking",
+        compute="_compute_picking_ids",
+        string="Generated stock pickings",
+        readonly=True,
+    )
+
+    picking_count = fields.Integer(
+        string="Picking count", compute="_compute_picking_ids", readonly=True
+    )
+
+    @api.depends("procurement_group_id.stock_move_ids.picking_id")
+    def _compute_picking_ids(self):
+        for record in self:
+            record.picking_ids = record.procurement_group_id.stock_move_ids.picking_id
+            record.picking_count = len(record.picking_ids)
+
+    def action_run(self):
+        if self.procurement_group_id:
+            return self.procurement_group_id
+
+        self.procurement_group_id = self.env["procurement.group"].create(
+            self._get_procurement_group_parameters()
+        )
+
+        quant_groups = self._get_existing_quants()
+
+        procurements = []
+
+        for line in self.stock_resupply_order_line_ids:
+            # service products have no tracking/lot and cannot be run in procurement
+            if line.product_id.product_tmpl_id.type == "service":
+                continue
+
+            available_quantity = self._get_available_quantity_for_product(
+                quant_groups, line
+            )
+
+            if available_quantity < line.quantity:
+                procurements.append(
+                    self.env["procurement.group"].Procurement(
+                        product_id=line.product_id,
+                        product_qty=line.quantity - available_quantity,
+                        product_uom=line.product_id.product_tmpl_id.uom_id,
+                        location_id=self.location_id,
+                        name=f"Resupply Order to {self.location_id.name}",
+                        origin=f"Resupply Order to {self.location_id.name}",
+                        company_id=self.company_id,
+                        values=self._get_procurement_values(),
+                    )
+                )
+
+        if procurements:
+            self.env["procurement.group"].run(procurements)
+
+        self.state = "done"
+
+        return self.procurement_group_id
+
+    @api.model
+    def _get_available_quantity_for_product(
+        self, quant_groups, stock_resupply_order_line
+    ):
+        try:
+            # I could not find a way to merge stock_resupply_order_line_ids
+            # with the quant_groups query, so it is retrieved with a product
+            # id search here. Not ideal, but it works.
+            group = next(
+                group
+                for group in quant_groups
+                if group["product_id"][0] == stock_resupply_order_line.product_id.id
+            )
+
+            return group["quantity"] - group["reserved_quantity"]
+        except StopIteration:
+            return 0
+
+    def _get_procurement_group_parameters(self):
```

Comment:
```suggestion
    def _prepare_procurement_group_values(self):
```

Following previous change request